Developer-Guide.md

```diff
@@ -273,22 +291,199 @@ $ popd
+
+## Update Docker configuration
 
+```bash
+$ sudo mkdir -p /etc/systemd/system/docker.service.d/
+$ cat <<EOF | sudo tee /etc/systemd/system/docker.service.d/kata-containers.conf
+[Service]
+ExecStart=
+ExecStart=/usr/bin/dockerd -D --add-runtime kata-runtime=/usr/bin/kata-runtime --default-runtime=kata-runtime
+EOF
 ```
-$ dir=/etc/systemd/system/docker.service.d
-$ file="$dir/kata-containers.conf"
-$ sudo mkdir -p "$dir"
-$ sudo test -e "$file" || echo -e "[Service]\nType=simple\nExecStart=\nExecStart=/usr/bin/dockerd -D --default-runtime runc" | sudo tee "$file"
-$ sudo grep -q "kata-runtime=" $file || sudo sed -i 's!^\(ExecStart=[^$].*$\)!\1 --add-runtime kata-runtime=/usr/local/bin/kata-runtime!g' "$file"
+
+## Restart the Docker systemd service
+
+```bash
 $ sudo systemctl daemon-reload
 $ sudo systemctl restart docker
 ```
 
-# Create a Kata Container
+## Run Kata Containers container
+
+You are now ready to run Kata Containers container:
+
+```
+$ sudo docker run -ti busybox sh
+```
+
+# If you want to run Kata Containers with Kubernetes
+
+Now that Kata Containers has been installed on your system, you need some
+extra components on top to make this work with Kubernetes.
+
+## Install CNI
+
+Kubernetes networking relies on CNI plugins, meaning you have to install
+them and choose which one will fit your needs. Follow the documentation
+[here](https://github.com/containernetworking/cni#running-the-plugins) and
+you can also refer to our CI script
+[here](https://github.com/kata-containers/tests/blob/master/.ci/install_cni_plugins.sh)
+if you are interested in understanding the full setup.
+
+## Install a CRI implementation
+
+Kata Containers runtime is an OCI compatible runtime and cannot directly
+interact with the CRI API level. For this reason, we need to rely on a
+CRI implementation translating CRI into OCI. There are two supported ways
+called [CRI-O](https://github.com/kubernetes-incubator/cri-o) and
+[CRI-containerd](https://github.com/containerd/cri). It is up to you to
+choose the one that you want, but you have to pick one, and make appropriate
+changes to make sure they will rely on Kata Containers runtime.
+
+### CRI-O
+
+If you picked CRI-O, please follow the instructions
+[here](https://github.com/kubernetes-incubator/cri-o/blob/master/tutorial.md)
+to get it properly installed.
+
+Now, the idea is that you will have to modify CRI-O configuration with
+some information about different container runtimes. By default, we will
+choose `runc`, but we also specify another container runtime, Kata Containers
+runtime in this case, as the runtime to run _untrusted_ workloads. In other
+words, this defines an alternative runtime that will be picked when the
+workload cannot be trusted and a higher level of security is required.
+An additional flag can be used to let CRI-O know if a workload should be
+considered _trusted_ or _untrusted_ by default.
+
+Additionally, we need the network namespace management to be performed by
```

Comment:
Lines 361-363 suggested rewrite: Additionally, we need CRI-O to perform the network namespace management. Otherwise, when the VM starts the network is not set and it will not get connected later.

Reply:
done